src/EffectTransform.js

```diff
@@ -20,95 +20,104 @@ const CENTER_X = 0.5;
  */
 const CENTER_Y = 0.5;
 
-// color conversions grabbed from https://gist.github.com/mjackson/5311256
+/**
+ * Reused memory location for storing an HSV color value.
+ * @type {Array<number>}
+ */
+const __hsv = [0, 0, 0];
 
 /**
- * Converts an RGB color value to HSL. Conversion formula
- * adapted from http://en.wikipedia.org/wiki/HSL_color_space.
- * Assumes r, g, and b are contained in the set [0, 255] and
- * returns h, s, and l in the set [0, 1].
+ * Converts an RGB color value to HSV. Conversion formula
+ * adapted from http://lolengine.net/blog/2013/01/13/fast-rgb-to-hsv.
+ * Assumes r, g, and b are in the range [0, 255] and
+ * returns h, s, and v in the range [0, 1].
  *
- * @param   {number}  r       The red color value
- * @param   {number}  g       The green color value
- * @param   {number}  b       The blue color value
- * @return  {Array}           The HSL representation
+ * @param   {Array<number>} rgb   The RGB color value
+ * @param   {number}        rgb.r The red color value
+ * @param   {number}        rgb.g The green color value
+ * @param   {number}        rgb.b The blue color value
+ * @param   {Array<number>} dst   The array to store the RGB values in
+ * @return  {Array<number>}       The `dst` array passed in
  */
-const rgbToHsl = ([r, g, b]) => {
-    r /= 255;
-    g /= 255;
-    b /= 255;
-
-    const max = Math.max(r, g, b);
-    const min = Math.min(r, g, b);
-    let h;
-    let s;
-    const l = (max + min) / 2;
-
-    if (max === min) {
-        h = s = 0; // achromatic
-    } else {
-        const d = max - min;
-        s = l > 0.5 ? d / (2 - max - min) : d / (max + min);
-
-        switch (max) {
-        case r: h = ((g - b) / d) + (g < b ? 6 : 0); break;
-        case g: h = ((b - r) / d) + 2; break;
-        case b: h = ((r - g) / d) + 4; break;
-        }
+const rgbToHsv = ([_r, _g, _b], dst) => {
```

Comment:
Since the `_` versions of these variables aren't being used independently I think it's OK to stick to the old destructuring parameter assignment.
```suggestion
const rgbToHsv = ([r, g, b], dst) => {
```